### ISSUE TITLE
Plan of work

### ISSUE THREAD

**Issue by kaplun** [inspirehep/invenio-orcid#1]
As a continuation from inspirehep/inspire-next#1093 (quoting @jirikuncar):
[...]
> What records would need to be sent to ORCID (and would therefore trigger an update/insert (in our case any record belonging to Literature and Data.

* [ ] It can be defined as a query to which a record has to match.

> An algorithm to retrieve which author with an ORCID are involved for the affected record (this is the tricky part: in our case we are retrieving all the signature from a paper, then we consider all the claimed ones, then we are following up into the corresponding authority record and retrieve the ORCID from there.

* [ ] Installations can provide `extract_authors` function. For records coming from deposit, it can be pretty straight forward.

> A way to retrieve from the ORCID the given token (or maybe we can just specify here the name of the oauth service used, in our case orcid.

* [ ] It can be done through following relation: Author -> User -> Linked OAuth account (ORCID). You can see an example in Invenio-GitHub.

> A mapping from the inner schema to the ORCID specific one (we have developed our internal dojson mapping, for that.

* [ ] One can configure a serializer as it is done in Invenio-Records-REST.

You specific implementation can be in INSPIRE overlay the same way as Zenodo is doing.
[...]

**Comment by jirikuncar** [comment]
@kaplun what about moving it under @inveniosoftware so everybody can find it?

**Comment by kaplun** [comment]
For control: we would need that it first matches our requirements and in this way we can control when we can push to it, what to push to it and when to make releases. When it reaches stability we can afterwords move it to @inveniosoftware.

Meanwhile it will be already findable through pypi.

**Comment by tiborsimko** [comment]
@kaplun Each repository under `inveniosoftware` organisation umbrella can have its own team of maintainers, so it should not be a problem to agree upon and create a new team who has push rights to `invenio-orcid`. Also, the deployment on PyPI can be automated, most of the packages have this already, so that it is sufficient to push a new tag and a PyPI release is created automatically.  

Hence I guess these technical considerations shouldn't be stoppers... I guess @jirikuncar was wondering about the questions of the wider Invenio ecosystem regarding visibility and reusability.

Basically, I'd say that an `invenio-foo` package can live anywhere, but if it wants to live in the central place under the `inveniosoftware` organisation umbrella, then it has to follow basic Invenio collaboration principles we have agreed upon, such as "each PR should increase or preserve code coverage", or "each PR should be seen by another pair of eyes", etc.  See the recap in https://indico.cern.ch/event/441770/contributions/1093979/attachments/1171100/1690728/iugw2015-invenio-development-practices.pdf

Another rule of thumb is include the needs of other services into account, e.g. `invenio-foo` should be pretty generic and willing to include other use cases, while `zenodo-foo` or `inspirehep-foo` can stay pretty service specific. For example, the GitHub integration was previously living inside Zenodo, now it has its own `invenio-github` canonical place, so that anyone can profit.

We can discuss these during one of the next Invenio Triage meetings, if we are ready to resume them... perhaps after Zenodo 3 is released.

**Comment by kaplun** [comment]
> Basically, I'd say that an invenio-foo package can live anywhere, but if it wants to live in the central place under the inveniosoftware organisation umbrella, then it has to follow basic Invenio collaboration principles we have agreed upon, such as "each PR should increase or preserve code coverage", or "each PR should be seen by another pair of eyes", etc.

Ahem... but it "didn't want to live there". I was precisely creating the repo under inspirehep umbrella in order to have (initial) relaxed constraints (as GitHub integration was initially on Zenodo). Of course there's no need for `inspirehep-orcid` (since we are pretty happy having the current code in our overlay), while `invenio-orcid` is the natural evolution of this work of ours.

Anyway these are all matters of principle.

I haven't appreciated having the code kidnapped, but I agree on centralizing the code under inveniosoftware, taking into account other services needs, as long as the INSPIRE service (@inspirehep/core) can retain control of it, so as to allow fostering our pragmatic service oriented policies.

**Comment by jirikuncar** [comment]
> I haven't appreciated having the code kidnapped, ...

No problem with removing it if you feel like that.

**Comment by kaplun** [comment]
It's fine: politics is the art of compromises :+1:

**Comment by kaplun** [comment]
> We can discuss these during one of the next Invenio Triage meetings, if we are ready to resume them... perhaps after Zenodo 3 is released.

Yes, would be great to get again regularly together in the same room and synchronize on all the ongoing works.

**Comment by jirikuncar** [comment]
Cool, if you don't insist on removing it then let's continue the effort on making the Invenio-ORCID package community friendly so all instances can profit from our joined efforts 👍

**Comment by audub** [comment]
We would also very much appreciate participating in some sort of "in the same room sync". I believe it is important to streamline efforts and also for everyone to stay up to date on what is happening.

**Comment by kaplun** [comment]
@jirikuncar, sure, just grant @inspirehep/core admin rights as well and I won't "insist" :wink: (if you fear this is too many people, whitelist at least me, @jmartinm and @jacquerie)

**Comment by jirikuncar** [comment]
@kaplun all @inveniosoftware/triagers have write access to all (most) @inveniosoftware repositories. I would like to remind http://invenio.readthedocs.io/en/latest/processes/index.html (do not merge your own or team mate PR, wait for cross service agreement, etc.) - we can complete it with @tiborsimko when triage meetings are revived.

**Comment by kaplun** [comment]
@jirikuncar thanks. When I believe when I verified the other day I didn't have rights, but maybe I haven't seen them.

Regarding the process I would like indeed to propose some amendements, to give it a pragmatic twist, e.g. for the early stages of components (i.e. when the adoption is still low, and correspondingly the risks are low).

But we can discuss in details indeed during triage.
